scipy/stats/stats.py

```diff
@@ -557,12 +557,12 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
     10.0
 
     """
-    a = asarray(a)
+    a = asarray(a, dtype=np.float64)
```

Comment:
Is this necessary? Shouldn't it work on at least `float32` and maybe also `float16` and not need to convert?

Reply:
You're right. `np.mean` and `am.mean` cast where necessary.